### DISCUSSION TITLE
Unable to overwrite k3s agent configuration.

### DISCUSSION THREAD

**Discussion by Puhanshibani** [k3s-io/k3s#7189] [Q&A]
**Environmental Info:**
k3s master is running in my local machine and i am trying to run the k3s agent in edge device but there i dont have root access along with can't write anything to  /etc, /var, /usr/bin directory. So k3s agent not able write the agent configuration to /etc/rancher directory. While running k3s agent it is not able to find out the default configuration.  **Is there any way to write the agent configuration into any custom directory instead of /etc/rancher.**

K3s Version: 

1.26

Node(s) CPU architecture, OS, and Version: 

Cluster Configuration:

1 master, 2 agents

**Describe the bug:**

Not able write the agent configuration to custom directory other than  /etc/rancher directory.

**Steps To Reproduce:**

- Installed K3s on edge device

**Expected behavior:**

ks3 agent should connect to mster.

**Actual behavior:**

![k3s-agent_1](https://user-images.githubusercontent.com/43839320/229024912-500ecbb4-5d7d-4f4d-a8f8-1bb51020198b.png)

**Additional context / logs:**

Is there any way to overwrite the agent configuration into any custom directory instead of /etc/rancher.

**Comment by brandond** [comment]
No. Can you bind-mount a writable partition to /etc/rancher?

Note that you will also need to run the agent as root... rootless agents are not supported.
